src/metabase/driver/h2.clj

```diff
@@ -121,29 +121,47 @@
       (when (instance? SessionLocal session)
         (Parser. session)))))
 
-(defn- contains-ddl?
-  [database query]
+(defn- check-disallow-ddl-commands
+  [{:keys [database] {:keys [query]} :native}]
   (when-let [h2-parser (make-h2-parser database)]
-    (try
-      (let [command      (.prepareCommand h2-parser query)
-            command-type (.getCommandType command)]
-        ;; TODO: do we need to handle CommandList?
-        ;; Command types are organized with all DDL commands listed first
-        ;; see https://github.com/h2database/h2database/blob/master/h2/src/main/org/h2/command/CommandInterface.java
-        (< command-type CommandInterface/ALTER_SEQUENCE))
-      ;; if the query is invalid, then it isn't DDL
-      (catch Throwable _ false))))
-
-(defn- check-disallow-ddl-commands [{:keys [database] {:keys [query]} :native}]
-  (when (and query (contains-ddl? database query))
-    (throw (IllegalArgumentException. "DDL commands are not allowed to be used with h2."))))
+    (when-let [command (try (.prepareCommand h2-parser query)
+                            ;; if the query is invalid, errors will get caught later
+                            (catch Throwable _ nil))]
+      (let [command-type (.getCommandType command)]
```

Comment:
[The javadoc](https://javadoc.io/static/com.h2database/h2/2.1.212/org/h2/command/CommandInterface.html#getCommandType--) says this returns an `int` (that you would then compare with constants they define, e.g. `CommandInterface.SELECT` is for a select statement).

Reply:
Yes, I'm comparing with `CommandInterface.SELECT`. The `let` and `cond` were a bit funky before though, so I've refactored them https://github.com/metabase/metabase/pull/28212/commits/516ad7c4b9e910025018d54e44bc5da66aaf55c5

Reply:
I deleted the whole function, see comment below